go/libraries/doltcore/schema/alterschema/addcolumn.go

```diff
@@ -83,23 +83,36 @@ func updateTableWithNewSchema(ctx context.Context, tbl *doltdb.Table, tag uint64
 		return nil, err
 	}
 
-	if defaultVal == nil {
+	if defaultVal == "" {
 		return doltdb.NewTable(ctx, vrw, newSchemaVal, rowData, &indexData)
 	}
 
 	me := rowData.Edit()
 
+	newSqlSchema, err := sqleSchema.FromDoltSchema(tblName, newSchema)
+	if err != nil {
+		return nil, err
+	}
+	columnIndex := []int{-1}
```

Comment:
Why is this a slice instead of just an int?

Reply:
Because `ApplyDefaults` wants a slice, so I just made it here. I could have this be `columnIndex := -1`, and then later do `ApplyDefault(..., []int{columnIndex}, ...)`